docs/architecture/adr-043-nft-module.md

```diff
@@ -0,0 +1,152 @@
+
+Considering generic usage and compatibility of interchain protocols including IBC and Gravity Bridge, it is preferred to have a very simple NFT module design which only stores NFTs by id and owner. 
+
+This design idea can enable composability that application-specific functions should be managed by other modules on Cosmos Hub or on other Zones by importing the NFT module.
+
+For example, NFTs can be grouped into collections in a collectibles module to define the behaviors such as minting, burning, etc.
+
+The current design is based on the work done by [IRISnet team](https://github.com/irisnet/irismod/tree/master/modules/nft) and an older implementation in the [Cosmos repository](https://github.com/cosmos/modules/tree/master/incubator/nft).
+
+
+## Decision
+
+We will create a module `x/nft` which only stores NFTs by id and owner.
+
+```proto
+message NFT {
+  string id    = 1;
+  string owner = 2;
+  google.protobuf.Any data = 3;
+}
+```
+
+The NFT conforms to the following specifications:
+  * The Id is an immutable field used as a unique identifier. NFT identifiers don't currently have a naming convention but
+    can be used in association with existing Hub attributes, e.g., defining an NFT's identifier as an immutable Hub address allows its integration into existing Hub account management modules. 
+    We envision that identifiers can accommodate mint and transfer actions.
+    The Id is also the primary index for storing NFTs.
+    ```
+    id (string) --> NFT (bytes)
+    ```
+  * Owner: This mutable field represents the current account owner (on the Hub) of the NFT, i.e., the account that will have authorization
+    to perform various activities in the future. This can be a user, a module account, or potentially a future NFT module that adds capabilities.
+    Owner is also the secondary index for storing NFT IDs owned by an address
+    ```
+    owner (address) --> []string
+    ```
+  * Data: This mutable field allows attaching special information to the NFT, for example:
+    * metadata such as title of the work and URI
+    * immutable data and parameters (such actual NFT data, hash or seed for generators)
+    * mutable data and parameters that change when transferring or when certain criteria are met (such as provenance)
+      
+    This ADR doesn't specify values that this field can take; however, best practices recommend upper-level NFT modules clearly specify its contents.
+    Although the value of this field doesn't provide additional context required to manage NFT records, which means that the field can technically be removed from the specification, 
+    the field's existence allows basic informational/UI functionality.
+
+The logic for transferring the ownership of an NFT to another address (no coins involved) should also be defined in this module
+```go
+func (k Keeper) TransferOwnership(nftID string, currentOwner, newOwner sdk.AccAddress) error
```

Comment:
We only define this method in the keeper to execute general logic without a `Msg`, the `Msg` should be defined in the upper-level/app-specific specific module.

Reply:
With Protobuf service oriented architecture (which we are using since Stargate), we are exposing module interfaces through Protobuf services. Keeper is rather used as an internal data access layer. 

So, since Stargate we want to use Proto definitions for module interface, and Keeper if you want to talk about internal data layout in the store.
See ADR-30 (authz module) for an example.

Reply:
ADR-33 is still in active development. I think ADR-30 will give you a good example. Also, as mentioned in other comment, for events please look at ADR-32